src/Data/Char/Base.agda

```diff
@@ -47,6 +48,10 @@ infix 4 _<_
 _<_ : Rel Char zero
 _<_ = ℕ._<_ on toℕ
 
+infix 4 _≤_
+_≤_ : Rel Char zero
+_≤_ = ReflClosure _<_
```

Comment:
Hmm so there are many different ways to define this:
```agda
1) _≤_ = ReflClosure _<_
2) _≤_ = ℕ._<_ on toℕ
3) _≤_ = StrictToNonStrict _<_
```
We already use 2) in `Data.Fin.Base` so it seems to me we should be consistent unless there are excellent reasons not to?

While we're here and discussing the differences between 1) and 3), what are the differences between `_≡_` and `_≈_`? It would be nice to have a bit of explanatory text added to the definition of `_≈_`.

Reply:
With `2) _≤_ = ℕ._<_ on toℕ` I assume you mean `2) _≤_ = ℕ._≤_ on toℕ`.
I used `ReflClosure` because that is literally how I was thinking about it:
the reflexive closure of the strict order.

> what are the differences between `_≡_` and `_≈_`?

`_≈_` has worse type reconstruction (which made me consider wrapping `_≤_`
in a record)? :speak_no_evil: 
I did not reallly want to pick one over the other (cf. the two versions of
`<-cmp` for instance) but perhaps it's not that big of a deal given they are
equivalent?

Reply:
> With 2) _≤_ = ℕ._<_ on toℕ I assume you mean 2) _≤_ = ℕ._≤_ on toℕ.

Yes!

> _≈_ has worse type reconstruction (which made me consider wrapping _≤_ in a record)?

If `_≈_` is equivalent, but has worse type reconstruction, why do we have it?

Reply:
Because until we had primitive proofs about `toℕ` being injective it was
impossible to prove that it was equivalent. :D

Reply:
So should we deprecate it?

Reply:
done